bin/replication.js

```diff
@@ -99,25 +109,28 @@ class _SetupReplication {
         this._log = log;
         this._sourceBucket = sourceBucket;
         this._targetBucket = targetBucket;
-        // TODO: support bootstrap list failover
-        const destEndpoint = destination.bootstrapList[0].servers[0];
+        this.destEndpoint =
+            new RoundRobin(destination.bootstrapList[0].servers);
         const verifySourceProfile = sourceProfile === undefined ?
             'default' : sourceProfile;
         const verifyTargetProfile = targetProfile === undefined ?
             'default' : targetProfile;
-
+        source.auth.vault.host = 'localhost';
+        source.s3.host = 'localhost';
+        const destHost = this.destHosts.pickHost().host;
         this._s3Clients = {
             source: _setupS3Client(source.transport,
                 `${source.s3.host}:${source.s3.port}`,
                 verifySourceProfile),
-            target: _setupS3Client(destination.transport, destEndpoint,
+            target: _setupS3Client(destination.transport, destHost,
```

Comment:
So the idea is that for destination it is sending request directly to the s3-nginx which runs on port 80. I will do a followup PR later where we could calculate the signature using a proxy path.